src/event_manager/control_ops.rs

```diff
@@ -0,0 +1,78 @@
+use crate::epoll::ControlOperation;
+
+use super::epoll_context::EpollContext;
+use super::{Error, Events, Result, SubscriberId};
+
+/// This opaque object is associated with an `EventSubscriber`, and allows the addition,
+/// modification, and removal of file descriptor events within the inner epoll set of an
+/// `EventManager`.
+// Right now this is a concrete object, but going further it can be turned into a trait and
+// passed around as a trait object.
+pub struct ControlOps<'a> {
```

Comment:
Why don't we simply make the association between the EpollContext and the subscriber ID part of EventManager directly? That would remove the indirection layer and the opaque object.

Reply:
Can you please provide a very short example here? (for example, how do you see the signature of any method impacted by your suggestion changing with the removal of the indirection layer). I have a couple of different things in mind that kinda fit your description, but I'm not quite sure which one actually matches.